web/modules/custom/dpl_login/dpl_login.module

```diff
@@ -199,21 +198,22 @@ function dpl_login_form_openid_connect_admin_settings_alter(array &$form, FormSt
  *
  * @throws \Drupal\dpl_login\Exception\MissingConfigurationException
  */
-function dpl_login_openid_connect_admin_settings_submit(array &$form, FormStateInterface $form_state): void {
+function dpl_login_openid_connect_admin_settings_validate(array &$form, FormStateInterface $form_state): void {
   $messenger = \Drupal::messenger();
 
   /** @var \Drupal\dpl_login\Adgangsplatformen\Config $adgangsplatformen_config */
   $adgangsplatformen_config = \Drupal::service('dpl_login.adgangsplatformen.config');
   $agency_id = $adgangsplatformen_config->getAgencyId();
+  $values = $form_state->getValues();
 
   // If agency_id is different from the already stored, we want to
   // force a new token to be created that belongs to the new agency.
-  if ($form_state->getValue('agency_id') != $agency_id) {
+  if ($values['clients']['adgangsplatformen']['settings']['agency_id'] != $agency_id) {
```

Comment:
What is the difference here?

Reply:
The difference is that if I am just calling `$form_state->getValue('agency_id')` it is always returning NULL. For some reason I have to explicitly go through the array to get the agency_id value.